packages/webview_flutter/webview_flutter/ios/Classes/FlutterWebView.h

```diff
@@ -7,26 +7,28 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/**
+ * The WkWebView used for the plugin.
+ *
+ * This class overrides some methods in `WKWebView` to serve the needs for the plugin.
+ */
+@interface FLTWKWebView : WKWebView
+@end
+
 @interface FLTWebViewController : NSObject <FlutterPlatformView, WKUIDelegate>
 
 - (instancetype)initWithFrame:(CGRect)frame
                viewIdentifier:(int64_t)viewId
                     arguments:(id _Nullable)args
               binaryMessenger:(NSObject<FlutterBinaryMessenger>*)messenger;
 
+- (instancetype)initWithWebView:(FLTWKWebView*)webView;
+
```

Comment:
Same as with Android, creating a constructor just to facilitate unit-testing is a bad practise. On the iOS side I haven't made any refactor changes so have a look if you can solve it in a similar way.